manifests/0000_90_kube-controller-manager-operator_05_alerts.yaml

```diff
@@ -39,3 +39,27 @@ spec:
           for: 15m
           labels:
             severity: critical
+        - alert: GarbageCollectorSyncFailed
+          annotations:
+            summary: There was a problem with syncing the resources for garbage collection.
+            description: Garbage Collector had a problem with syncing and monitoring the available resources. Please see KubeControllerManager logs for more details.
+          expr: |
+            rate(garbagecollector_controller_garbagecollector_resources_sync_error_total{}[1h]) > 0
+          labels:
+            severity: warning
+        - alert: GarbageCollectorObjectDeletionFailed
+          annotations:
+            summary: There was a problem with deleting an object.
+            description: Garbage Collector had some problems when attempting to delete an object. Please see KubeControllerManager logs for more details.
+          expr: |
+            rate(workqueue_retries_total{name="garbage_collector_attempt_to_delete"}[5m]) > 0
```

Comment:
This might be very noisy. The alert will fire whenever a retry is detected. I am not sure how common it is for the GC, but in some other contexts, it is very normal to retry. Having a `for` clause here might make the alert more resilient to problems that are being resolved right away.

In general, warning level alerts, tend to have less priority for cluster admins, so waiting for even an hour before looking into one is pretty normal so as developers writing alerts we should thrive to reduce the noise for them and eliminate any kind of false positive.

Reply:
Yes it is insufficient and manifests problems you are describing. I have been trying to tweak these two (`garbage_collector_attempt_to_delete` and `garbage_collector_attempt_to_delete` ) and see how they perform in CI.

Ultimately I found out we probably have to bring in a different solution instead of this one: https://github.com/openshift/cluster-kube-controller-manager-operator/pull/623#discussion_r879920985